doc/sphinx/addendum/implicit-coercions.rst

```diff
@@ -167,25 +167,38 @@ Declaring Coercions
      Silence the non uniform inheritance warning.
 
   .. exn:: @qualid not declared.
-     :undocumented:
+
+     The given :token:`qualid` is not a defined constant.
 
   .. exn:: @qualid is already a coercion.
-     :undocumented:
+
+     The given :token:`qualid` is already registered a coercion.
 
   .. exn:: Funclass cannot be a source class.
-     :undocumented:
+
+     Funclass as a source class is not supported.
 
   .. exn:: @qualid is not a function.
-     :undocumented:
+
+     The given :token:`qualid` is not a function, so it cannot be used as a
+     coercion.
 
   .. exn:: Cannot find the source class of @qualid.
-     :undocumented:
+
+     A valid source class was unable to be inferred.
```

Comment:
```suggestion
     Coq can not infer a valid source class.
```

Hopefully we explain how things are inferred nearby...

Reply:
Well Coq just looks at the shape of the term and decides which sortclass it must be. When you declare a coercion `A >-> B` for `f : C -> D`, Coq will check the type of `f`, try to match `C` to the sortclass `A` and `B` to the sortclass `B`. Say that `B` was `Funclass`, then Coq would try to unfold `D` to check it actually is a function etc.

Reply:
I don't think class inference/checking does any unfolding

Reply:
Indeed, it doesn't. The "inference" is in fact a very direct matching.